yo1k/tic_tac_toe/core/game.py

```diff
@@ -0,0 +1,296 @@
+from enum import Enum, auto
+from collections.abc import Sequence
+from typing import Optional
+from abc import ABC, abstractmethod
+
+
+class Mark(Enum):
+    X = auto()
+    O = auto()
+
+
+class Player:
+    def __init__(self, mark: Mark):
+        self.__mark = mark
+        self.__wins = 0
+
+    def mark(self) -> Mark:
+        return self.__mark
+
+    @property
+    def wins(self) -> int:
+        return self.__wins
+
+    @wins.setter
+    def wins(self, wins: int):
+        self.__wins = wins
+
+    def __str__(self):
+        return f"{type(self)}(" \
+               f"mark={self.__mark}," \
+               f"wins={self.__wins})"
+
+
+class Phase(Enum):
+    ONSET = auto()
```

Comment:
[optional]
This name keeps bugging me:) If `BEGINNING` sounds not worse than `ONSET` for you, then please consider changing the name. Otherwise ignore.

Reply:
Your state of mind is important to me ;)
Done.

Reply:
This part is relatively non-trivial, and will be addressed in a separate PR.